### DISCUSSION TITLE
Calculating amounts for adding liquidity

### DISCUSSION THREAD

**Discussion by ibrahimkecici** [atiselsts/uniswap-v3-liquidity-math#7] [Q&A] [answered]
Hello. I need your help for this. I have 100 ETH and 10000 USDC. I want to add liquidity for ETH/USDC at range 1000-1600. Some ETH should be swapped to USDC. How can I calculate required ETH and USDC amounts for adding liquidity and required swap amount for ETH to USDC?

**Comment by atiselsts** [comment, marked as answer]
Let me answer a simpler question: what should the proportion of ETH and USDC be in the position?

To compute the proportion you can rely on the fact that liquidity in both sides of the position should be equal: `L_x = L_y`
Following the naming conventions from the Python code, the relevant liquidity math formulas are:

    L_y = y / (sp - sa) = x * sp * sb / (sb - sp) = L_x

Expressing `y` from the above:

    y = x * sp * sb *  (sp - sa) / (sb - sp)

Assuming x = 1, you can compute `y_unit`, that is, how much USDC is needed per one ETH.

As a side note, I can't believe that there still isn't a calculator for this online, after 1.5 years since v3 was released!


If you know this proportion, that is, how much USDC (Y) you need for one ETH (X), then knowing the current price P you can compute the value V of one unit (this is not checked, hope there are no mistakes):

    V_unit = 1 * P + y_unit
    x0 = 100; y0 = 10000
    V_total = x0 * P + y0
    N_units = V_total / V_unit
    x_final = N_units
    y_final = N_units * y_unit

Sell `x_0 - x_final` ETH to get to the right amount of both assets.

**Reply by ibrahimkecici** [reply]
Hey. Thanks for your answer. Also yes we are need a calculator. Maybe we can do it :)

**Reply by atiselsts** [reply]
If you or someone else reading this can help with the frontend dev, then I'm ready to help with the logic!
